File: tests/Web.AcceptanceTests/Pages/HistoryPage.cs
namespace LiftAndShift.Web.AcceptanceTests.Pages;

public class HistoryPage(IPage page) : BasePage(page)
{
    public override string PagePath => $"{BaseUrl}/history";

    public Task AssertHeading() => Assertions.Expect(Page.Locator("h1")).ToHaveTextAsync("Workout History");

    public Task AssertSessionVisible(string exerciseName) =>
        Assertions.Expect(Page.Locator("article", new() { HasText = exerciseName }).First).ToBeVisibleAsync();

    public Task AssertStatusTag(string exerciseName, string status) =>
        Assertions.Expect(Page.Locator("article", new() { HasText = exerciseName }).First.Locator(".status-tag"))
            .ToHaveTextAsync(status);

    public Task ClickViewFor(string exerciseName) =>
        Page.Locator("article", new() { HasText = exerciseName }).First.Locator("a:has-text('View')").ClickAsync();
}

File: src/Web/ClientApp/e2e/pages/history.page.ts
import { expect } from '@playwright/test';
import { BasePage, baseUrl } from './base-page';

export class HistoryPage extends BasePage {
  get pagePath(): string {
    return `${baseUrl()}/history`;
  }

  assertHeading() {
    return expect(this.page.locator('h1')).toHaveText('Workout History');
  }

  assertSessionVisible(exerciseName: string) {
    return expect(
      this.page.locator('article', { hasText: exerciseName }).first()
    ).toBeVisible();
  }

  assertStatusTag(exerciseName: string, status: string) {
    return expect(
      this.page.locator('article', { hasText: exerciseName }).first().locator('.status-tag')
    ).toHaveText(status);
  }

  clickViewFor(exerciseName: string) {
    return this.page
      .locator('article', { hasText: exerciseName })
      .first()
      .locator("a:has-text('View')")
      .click();
  }
}
